edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/requests/RestartRequestHandler.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Core.Requests
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public class RestartRequestHandler : RequestHandlerBase<RestartRequest, object>
+    {
+        static readonly Version ExpectedSchemaVersion = new Version("1.0");
+
+        readonly IEnvironmentProvider environmentProvider;
+        readonly IConfigSource configSource;
+        readonly ICommandFactory commandFactory;
+
+        public RestartRequestHandler(IEnvironmentProvider environmentProvider, IConfigSource configSource, ICommandFactory commandFactory)
+        {
+            this.environmentProvider = Preconditions.CheckNotNull(environmentProvider, nameof(environmentProvider));
+            this.configSource = Preconditions.CheckNotNull(configSource, nameof(configSource));
+            this.commandFactory = Preconditions.CheckNotNull(commandFactory, nameof(commandFactory));
+        }
+
+        public override string RequestName => "RestartModule";
+
+        protected override async Task<Option<object>> HandleRequestInternal(Option<RestartRequest> payloadOption, CancellationToken cancellationToken)
+        {
+            RestartRequest payload = payloadOption.Expect(() => new ArgumentException("Request payload not found"));
+            if (ExpectedSchemaVersion.CompareMajorVersion(payload.SchemaVersion, "logs upload request schema") != 0)
+            {
+                Events.MismatchedMinorVersions(payload.SchemaVersion, ExpectedSchemaVersion);
+            }
+
+            Events.ProcessingRequest(payload);
+
+            DeploymentConfigInfo deploymentConfigInfo = await this.configSource.GetDeploymentConfigInfoAsync();
+            IEnvironment environment = this.environmentProvider.Create(deploymentConfigInfo.DeploymentConfig);
+            ModuleSet modules = await environment.GetModulesAsync(cancellationToken);
+            if (!modules.TryGetModule(payload.Id, out IModule module))
+            {
+                throw new InvalidOperationException($"Module {payload.Id} not found in the current environment");
+            }
+
+            if (!(module is IRuntimeModule runtimeModule) || runtimeModule.RuntimeStatus != ModuleStatus.Running)
+            {
+                throw new InvalidOperationException($"Module {payload.Id} cannot be restarted since it is not running");
```

Comment:
I think if the module was stopped, I would expect this call to start it not error?

Reply:
Emm, I thought in that case we should let EdgeAgent start it (as per backoff policy, etc)?

Reply:
It does make sense to think of restarting as "Ensure it's stopped, then ensure it's started". So for an already stopped module it would just start it.

Reply:
Actually Mike and I agreed to not check the status at all.. simply invoke Restart on the module. That would be similar to what iotedge does.

Reply:
Cool. That should have the same behavior then (via `docker restart`).

Reply:
Should this if check be removed then?